streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java

```diff
@@ -134,143 +134,6 @@ public TopologyTestDriverTest(final boolean eosEnabled) {
-                return false;
-            }
-            final Record record = (Record) o;
-            return timestamp == record.timestamp &&
-                offset == record.offset &&
-                Objects.equals(key, record.key) &&
-                Objects.equals(value, record.value) &&
-                Objects.equals(topic, record.topic) &&
-                Objects.equals(headers, record.headers);
-        }
-
-        @Override
-        public int hashCode() {
-            return Objects.hash(key, value, headers, timestamp, offset, topic);
-        }
-    }
-
-    private final static class Punctuation {
-        private final long intervalMs;
-        private final PunctuationType punctuationType;
-        private final Punctuator callback;
-
-        Punctuation(final long intervalMs,
-                    final PunctuationType punctuationType,
-                    final Punctuator callback) {
-            this.intervalMs = intervalMs;
-            this.punctuationType = punctuationType;
-            this.callback = callback;
-        }
-    }
-
-    private final class MockPunctuator implements Punctuator {
-        private final List<Long> punctuatedAt = new LinkedList<>();
-
-        @Override
-        public void punctuate(final long timestamp) {
-            punctuatedAt.add(timestamp);
-        }
-    }
-
-    private final class MockProcessor implements Processor {
-        private final Collection<Punctuation> punctuations;
-        private ProcessorContext context;
-
-        private boolean initialized = false;
-        private boolean closed = false;
-        private final List<Record> processedRecords = new ArrayList<>();
-
-        MockProcessor(final Collection<Punctuation> punctuations) {
-            this.punctuations = punctuations;
-        }
-
-        @Override
-        public void init(final ProcessorContext context) {
-            initialized = true;
-            this.context = context;
-            for (final Punctuation punctuation : punctuations) {
-                this.context.schedule(Duration.ofMillis(punctuation.intervalMs), punctuation.punctuationType, punctuation.callback);
-            }
-        }
-
-        @Override
-        public void process(final Object key, final Object value) {
-            processedRecords.add(new Record(key, value, context.headers(), context.timestamp(), context.offset(), context.topic()));
-            context.forward(key, value);
-        }
-
-        @Override
-        public void close() {
-            closed = true;
-        }
-    }
-
-    private final List<MockProcessor> mockProcessors = new ArrayList<>();
-
-    private final class MockProcessorSupplier implements ProcessorSupplier {
-        private final Collection<Punctuation> punctuations;
-
-        private MockProcessorSupplier() {
-            this(Collections.emptySet());
-        }
-
-        private MockProcessorSupplier(final Collection<Punctuation> punctuations) {
```

Comment:
Here, we can schedule multiple punctuations with different types. Seems that `MockProcessor` only supports one punctuation... Is this intended?

Reply:
Yes that's true; though none of the tests actually schedules multiple punctuators, so I think it is not worth adding such feature (at least for now).